code/modules/mob/living/carbon/xenomorph/castes/hivelord/abilities_hivelord.dm

```diff
@@ -343,25 +343,33 @@ GLOBAL_LIST_INIT(thickenable_resin, typecacheof(list(
 
 	patient.add_filter("hivelord_healing_infusion_outline", 3, list("type" = "outline", "size" = 1, "color" = COLOR_VERY_PALE_LIME_GREEN)) //Set our cool aura; also confirmation we have the buff
 
+	var/datum/action/healing_infusion_storage_datum/storage_datum = new /datum/action/healing_infusion_storage_datum //Create the buff datum
+
 	patient.infusion_active = TRUE //Indicate the infusion as being active
 
-	health_ticks_remaining = HIVELORD_HEALING_INFUSION_TICKS
-	sunder_ticks_remaining = HIVELORD_HEALING_INFUSION_TICKS
+	storage_datum.health_ticks_remaining = HIVELORD_HEALING_INFUSION_TICKS
+	storage_datum.sunder_ticks_remaining = HIVELORD_HEALING_INFUSION_TICKS
 
-	addtimer(CALLBACK(src, .proc/healing_infusion_timer_check, patient), HIVELORD_HEALING_INFUSION_DURATION)
+	addtimer(CALLBACK(storage_datum, /datum/action/healing_infusion_storage_datum/.proc/healing_infusion_timer_check, patient), HIVELORD_HEALING_INFUSION_DURATION)
 
 	succeed_activate()
 	add_cooldown()
 
 	GLOB.round_statistics.hivelord_healing_infusions++ //Statistics
 	SSblackbox.record_feedback("tally", "round_statistics", 1, "hivelord_healing_infusions")
 
-	RegisterSignal(patient, COMSIG_XENOMORPH_HEALTH_REGEN, .proc/healing_infusion_regeneration) //Register so we apply the effect whenever the target heals
-	RegisterSignal(patient, COMSIG_XENOMORPH_SUNDER_REGEN, .proc/healing_infusion_sunder_regeneration) //Register so we apply the effect whenever the target heals
+	storage_datum.RegisterSignal(patient, COMSIG_XENOMORPH_HEALTH_REGEN, /datum/action/healing_infusion_storage_datum/.proc/healing_infusion_regeneration) //Register so we apply the effect whenever the target heals
+	storage_datum.RegisterSignal(patient, COMSIG_XENOMORPH_SUNDER_REGEN, /datum/action/healing_infusion_storage_datum/.proc/healing_infusion_sunder_regeneration) //Register so we apply the effect whenever the target heals
+
+
+///Store all relevant information for each target
+/datum/action/healing_infusion_storage_datum
```

Comment:
Why is this an action ?

Reply:
in case it needs to make owner references initially, but I don't think I ended up using them